source/textUtils/uniscribe.py

```diff
@@ -0,0 +1,40 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2024 NV Access Limited, Leonard de Ruijter
+
+"""Wrapper functions for NVDAHelper uniscribe functions."""
+
+import ctypes
+from typing import Generator
+
+
+def splitAtCharacterBoundaries(text: str) -> Generator[str, None, None]:
+	"""
+	Splits a given string into real visible characters (or glyphs), thereby respecting character boundaries.
+	Contrary to just iterating over a string, this respects surrogate pairs, decomposite characters, etc.
+	"""
+	import NVDAHelper  # Import late to avoid circular import.
+
+	if not NVDAHelper.localLib:
+		raise RuntimeError("NVDAHelper not initialized")
+	if not text:
+		return
+	# uniscribe does some strange things
+	# when you give it a string  with not more than two alphanumeric chars in a row.
+	# Inject two alphanumeric characters at the end to fix this

```

Comment:
should we be handling the end of this buffer somewhere when returning data? does this affect results

Reply:
The two characters are filtered out of the result.
The only impact this had, is that even when the `text` parameter was an empty string, the NVDAHelper function would still be called with just "xx". I fixed that by explicitly checking for an empty string.